haystack/nodes/preprocessor/splitter.py

```diff
@@ -0,0 +1,755 @@
+            split_length=split_length,
+            split_overlap=split_overlap,
+            max_chars=max_chars,
+            max_tokens=max_tokens,
+        )
+
+        if any(document.content_type != "text" for document in documents):
+            raise ValueError(
+                "Some documents do not contain text. Make sure to pass only text documents to this node. "
+                "You can use a RouteDocuments node to make sure only text documents are sent to the DocumentCleaner."
+            )
+
+        if split_by == "token" and not self.tokenizer:
+            raise ValueError(
+                "If you set split_by='token', you must give a value to 'tokenizer_model'. "
+                "Use the same model you're using for your Reader."
+            )
+
+        if split_by == "character":
+            splitter_function = lambda text: (text, [0] * len(text)) if text != "" else ([""], [0])
+
+        elif split_by == "regex":
+            if not split_regex:
+                raise ValueError("If 'split_by' is set to 'regex', you must give a value to 'split_regex'.")
+            splitter_function = partial(self.split_by_regex, pattern=split_regex)
+
+        elif split_by == "page":
+            splitter_function = lambda text: self.split_by_regex(text=text, pattern="\f")
+
+        elif split_by == "paragraph":
+            splitter_function = lambda text: self.split_by_regex(text=text, pattern="\n\n")
+
+        elif split_by == "sentence":
+            splitter_function = lambda text: self.split_by_sentence_tokenizer(text=text)
+
+        elif split_by == "word":
+            splitter_function = lambda text: self.split_by_regex(text=text, pattern="\s+")
+
+        elif split_by == "token":
+            if isinstance(self.tokenizer, TokenizerI):
+                splitter_function = lambda text: self.split_by_sentence_tokenizer(text=text)
+            else:
+                splitter_function = lambda text: self.split_by_transformers_tokenizer(text=text)
+        else:
+            raise ValueError("split_by must be either 'character', 'word', 'sentence', 'paragraph', 'page' or 'regex'")
+
+        final_documents = []
+        for document in tqdm(documents, disable=not self.progress_bar, desc="Splitting", unit="docs"):
+
+            # Split them into single unit documents
+            split_documents = self.split_into_units(
+                document=document, units=splitter_function(text=document.content), add_page_number=add_page_number
+            )[0]
+
+            # If we need to count the tokens, split it by token
+            if max_tokens:
+                for doc in split_documents:
+                    if isinstance(self.tokenizer, (PreTrainedTokenizer, FeatureExtractor)):
+                        tokens = self.split_by_transformers_tokenizer(text=doc.content)[0]
+                    elif isinstance(self.tokenizer, TokenizerI):
+                        tokens = self.split_by_nltk_word_tokenizer(text=doc.content)[0]
```

Comment:
Why are we using the nltk tokenizer here instead of regex tokenization?

Also, in case of word tokenization each doc will have one token in this case right? If we could skip it in that case then the tokenization would be faster.